docs/user_docs/concepts/intent

```diff
@@ -0,0 +1,90 @@
+
+### Rollback and Disaster Recovery
+
+We recommend to sotre the Intent in a version control systems like Git. This enables better change management practices and makes it simpler to roll back to previous versions if needed. In case of a failure or outage, having a validated Intent simplifies the rollback process, ensuring that the system can be quickly recovered.
+
+## Format
+
+The structure of Intent is shown as follow and we are keeping envolving it to meet more operational situations. A complete definition can be found [here](https://github.com/KusionStack/kusion/blob/main/pkg/models/intent.go).
+
+```yaml
+resources:
+- id: aliyun:alicloud:alicloud_db_instance:wordpress
+  type: Terraform
+  attributes:
+    engine: MySQL
+    engine_version: "5.7"
+    instance_storage: 20
+    instance_type: mysql.n2.serverless.1c
+    ... ...
+  extensions:
+    provider: registry.terraform.io/aliyun/alicloud/1.209.1
+    providerMeta:
+      region: cn-beijing-1
+    resourceType: alicloud_db_instance
+- id: apps/v1:Deployment:wordpress:wordpress-dev-wordpress
+  type: Kubernetes
+  attributes:
+    apiVersion: apps/v1
+    kind: Deployment
+    metadata:
+      labels:
+        app.kubernetes.io/name: wordpress
+        app.kubernetes.io/part-of: wordpress
+      name: wordpress-dev-wordpress
+      namespace: wordpress
+    spec:
+      replicas: 1
+      selector:
+        matchLabels:
+          app.kubernetes.io/name: wordpress
+          app.kubernetes.io/part-of: wordpress
+          ... ...
+  dependsOn:
+  - v1:Namespace:wordpress
+  - v1:Secret:wordpress:wordpress-db
+  - v1:Service:wordpress:wordpress-dev-wordpress-private
+  ... ...
+```
+
+The YAML above is an Intent file. Each item in the resources array represents an actually infrastructure resource like a K8s workload or a cloud resource. Currently, we only have a resoruce array in the Intent and we are going to add more components such as topologies and deploy strategies in it to cover the complete SDLC.
+
+The struct of the resource is shown as follow.
+
+```go
+	// ID is the unique key of this resource in the whole State.
+	// ApiVersion:Kind:Namespace:Name is an idiomatic way for Kubernetes resources.
+    // providerNamespace:providerName:resourceType:resourceName for Terraform resources
+	ID string `json:"id" yaml:"id"`
+
+	// Type represents all Runtimes we supported. e.g. Kubernetes, Terraform
+	Type Type `json:"type" yaml:"type"`
+
+	// Attributes represents all specified attributes of this resource
+	Attributes map[string]interface{} `json:"attributes" yaml:"attributes"`
+
+	// DependsOn contains all resources this resource depends on
+	DependsOn []string `json:"dependsOn,omitempty" yaml:"dependsOn,omitempty"`
+
+	// Extensions specifies arbitrary metadata of this resource
+	Extensions map[string]interface{} `json:"extensions,omitempty" yaml:"extensions,omitempty"`
+```
+
+## How it works
```

Comment:
Also this is not necessary, the [Architecture](https://www.kusionstack.io/docs/user_docs/concepts/arch) doc illustrate how this works.